storage/entities/peer.js

```diff
@@ -0,0 +1,306 @@
+			_.pick(this.defaultOptions, ['limit', 'offset', 'fieldSet'])
+		);
+
+		const params = Object.assign(
+			{},
+			{ limit: parsedOptions.limit, offset: parsedOptions.offset },
+			{ parsedFilters }
+		);
+
+		return this.adapter.executeFile(
+			{
+				[this.FIELD_SET_SIMPLE]: this.SQLs.selectSimple,
+			}[parsedOptions.fieldSet],
+			params,
+			{},
+			tx
+		);
+	}
+
+	/**
+	 * Create peer object
+	 *
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {*}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	create(data, options = {}, tx) {
+		const objectData = _.defaults(data, defaultCreateValues);
+		const createSet = this.getValuesSet(objectData);
+		const attributes = Object.keys(data)
+			.map(k => `"${this.fields[k].fieldName}"`)
+			.join(',');
+
+		return this.adapter.executeFile(
+			this.SQLs.create,
+			{ createSet, attributes },
+			{},
+			tx
+		);
+	}
+
+	/**
+	 * Update the records based on given condition
+	 *
+	 * @param {filters.Peer} [filters]
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {*}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	update(filters, data, options = {}, tx) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const objectData = _.omit(data, readOnlyFields);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+		const updateSet = this.getUpdateSet(objectData);
+
+		const params = Object.assign(objectData, {
+			parsedFilters,
+			updateSet,
+		});
+
+		return this.adapter.executeFile(this.SQLs.update, params, {}, tx);
+	}
+
+	/**
+	 * Update one record based on the condition given
+	 *
+	 * @param {filters.Peer} filters
+	 * @param {Object} data
+	 * @param {Object} [options]
+	 * @param {Object} tx - Transaction object
+	 * @return {*}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	updateOne(filters, data, options = {}, tx) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const objectData = _.omit(data, readOnlyFields);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+		const updateSet = this.getUpdateSet(objectData);
+
+		const params = Object.assign(objectData, {
+			parsedFilters,
+			updateSet,
+		});
+
+		return this.adapter.executeFile(this.SQLs.updateOne, params, {}, tx);
+	}
+
+	/**
+	 * Check if the record exists with following conditions
+	 *
+	 * @param {filters.Peer} filters
+	 * @param {Object} [options]
+	 * @param {Object} [tx]
+	 * @returns {Promise.<boolean, Error>}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	isPersisted(filters, options = {}, tx) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+
+		if (parsedFilters === '') {
```

Comment:
`validateFilters` method has an optional param `atLeastOneRequired` that will perform this check if set to true

Reply:
Done